### PR TITLE
feat: karma circle refactor ping-3529 ping-3530 ping-3525

### PR DESCRIPTION
**Pull request by yusufputra**
### Summary by CodeRabbit

- New Feature: Introduced a new prop `withKarma` to display the Karma score in the user profile header and discovery list.
- Refactor: Updated the `ProfilePicture` component to support additional props for size, width, and conditional rendering of Karma score.
- Style: Changed the color scheme of the Karma score indicator to shades of green for better visual appeal.
- Documentation: Enhanced the explanation text in the `KarmaScore` component to provide more clarity on how Karma score affects visibility and rewards on the platform.

### CODE REVIEW

**Comment by bastian-purrer** [comment]
Image description   CodeRabbit



Currently reviewing new changes in this PR...

<details>
<summary>Commits</summary>
Files that changed from the base of the PR and between df250e3c84c9bfe2b8d4e1d0b89590a912a3f29d and ad6692825369735e8b8a0a3752d3dc7645523edd commits.
</details>

<details>
<summary>Files selected (7)</summary>

* src/components/Karma/utils.ts (1)
* src/screens/DiscoveryScreenV2/elements/DiscoveryItemList.js (3)
* src/screens/DiscoveryScreenV2/fragment/UsersFragment.js (1)
* src/screens/OtherProfile/index.js (3)
* src/screens/ProfileScreen/elements/KarmaScore.tsx (2)
* src/screens/ProfileScreen/elements/ProfilePicture.js (2)
* src/screens/ProfileScreen/index.js (1)
</details>


<details>
<summary>Files ignored due to filter (1)</summary>

* ios/Podfile.lock

</details>





---

File: src/screens/DiscoveryScreenV2/fragment/UsersFragment.js [hunk 1]
```diff
@@ -172,6 +172,7 @@ const UsersFragment = ({
     ) {
       return (
         <DomainList
+          withKarma={true}
```
Comment (bastian-purrer) [inline src/screens/DiscoveryScreenV2/fragment/UsersFragment.js:175]: Image description   CodeRabbit

The `withKarma` prop is hardcoded to `true`. If this is intentional and you always want to display the karma, then it's fine. However, if you want to make it configurable, consider passing `withKarma` as a prop to the parent component.
```diff
- withKarma={true}
+ withKarma={this.props.withKarma}
```
This change will allow you to control the visibility of the karma score from the parent component.

File: src/screens/ProfileScreen/elements/ProfilePicture.js [hunk 2]
```diff
@@ -24,36 +27,55 @@ const ProfilePicture = ({
...
+              style={styles.profileImage(size, width)}
               source={{
                 uri: profilePicPath ? `${profilePicPath}` : DEFAULT_PROFILE_PIC_PATH
               }}
               resizeMode={FastImage.resizeMode.stretch}
             />
             {renderAddIcon()}
           </View>
-        </CircleGradient>
+        )}
```
Comment (bastian-purrer) [inline src/screens/ProfileScreen/elements/ProfilePicture.js:60]: Image description   CodeRabbit

The code is mostly fine, but there's a bit of redundancy in the rendering of the `Image` and `renderAddIcon()` components. These are rendered in both branches of the conditional statement, which can lead to code duplication and increased maintenance effort. Consider extracting this into a separate function.

```diff
-          {withKarma ? (
-            <CircleGradient
-              fill={karmaScore}
-              size={normalize(size)}
-              width={normalize(width)}
-              testId="images">
-              <View style={styles.profileImageContainer(size)}>
-                <Image
-                  testId="images"
-                  style={styles.profileImage(size, width)}
-                  source={{
-                    uri: profilePicPath ? `${profilePicPath}` : DEFAULT_PROFILE_PIC_PATH
-                  }}
-                  resizeMode={FastImage.resizeMode.stretch}
-                />
-                {renderAddIcon()}
-              </View>
-            </CircleGradient>
-          ) : (
-            <View style={styles.profileImageContainer(size)} testId="images">
-               <Image
-                testId="images"
-                style={styles.profileImage(size, width)}
-                 source={{
-                   uri: profilePicPath ? `${profilePicPath}` : DEFAULT_PROFILE_PIC_PATH
-                 }}
-                 resizeMode={FastImage.resizeMode.stretch}
-               />
-               {renderAddIcon()}
-             </View>
-          )}

+          {withKarma ? (
+            <CircleGradient
+              fill={karmaScore}
+              size={normalize(size)}
+              width={normalize(width)}
+              testId="images">
+              {renderProfileContent()}
+            </CircleGradient>
+          ) : renderProfileContent()}

// Add this new function
const renderProfileContent = () => (
  <View style={styles.profileImageContainer(size)} testId="images">
    <Image
      testId="images"
      style={styles.profileImage(size, width)}
      source={{
        uri: profilePicPath ? `${profilePicPath}` : DEFAULT_PROFILE_PIC_PATH
      }}
      resizeMode={FastImage.resizeMode.stretch}
    />
    {renderAddIcon()}
  </View>
);
```

**Review by bastian-purrer — COMMENTED** [review]
Image description   CodeRabbit

<details>
<summary>Commits</summary>
Files that changed from the base of the PR and between df250e3c84c9bfe2b8d4e1d0b89590a912a3f29d and ad6692825369735e8b8a0a3752d3dc7645523edd commits.
</details>

<details>
<summary>Files selected (7)</summary>

* src/components/Karma/utils.ts (1)
* src/screens/DiscoveryScreenV2/elements/DiscoveryItemList.js (3)
* src/screens/DiscoveryScreenV2/fragment/UsersFragment.js (1)
* src/screens/OtherProfile/index.js (3)
* src/screens/ProfileScreen/elements/KarmaScore.tsx (2)
* src/screens/ProfileScreen/elements/ProfilePicture.js (2)
* src/screens/ProfileScreen/index.js (1)
</details>


<details>
<summary>Files ignored due to filter (1)</summary>

* ios/Podfile.lock

</details>






<details>
<summary>Files skipped from review due to trivial changes (5)</summary>

* src/components/Karma/utils.ts
* src/screens/DiscoveryScreenV2/elements/DiscoveryItemList.js
* src/screens/OtherProfile/index.js
* src/screens/ProfileScreen/elements/KarmaScore.tsx
* src/screens/ProfileScreen/index.js

</details>

<details>
<summary>Review comments generated (4)</summary>

* Review: 2
* LGTM: 2

</details>

---

<details>
<summary>Tips</summary>

### Chat with Image description  CodeRabbit Bot (`@coderabbitai`)
- Reply on review comments left by this bot to ask follow-up questions. A review comment is a comment on a diff or a file.
- Invite the bot into a review comment chain by tagging `@coderabbitai` in a reply.

### Code suggestions
- The bot may make code suggestions, but please review them carefully before committing since the line number ranges may be misaligned. 
- You can edit the comment made by the bot and manually tweak the suggestion if it is slightly off.

### Pausing incremental reviews
- Add `@coderabbitai: ignore` anywhere in the PR description to pause further reviews from the bot.

</details>

**Review by alamsyaah — APPROVED** [review]

**Pull request merged**